client/src/components/inventory/HardwareList.tsx

```diff
@@ -77,41 +190,116 @@ export class HardwareList extends React.Component<Props, { isLoading: boolean }>
+                                        </Header>
+                                    </Accordion.Title>
+                                    <Accordion.Content
+                                        active={this.state.activeIndices.includes(cat.category.category_id)}>
+                                        <Item.Group>
+                                            {cat.items.filter((searchInput => this.containsSearchQuery(searchInput))).map((it: HwItem) => {
+                                                return (<HardwareItem
+                                                    item_name={it.item_name}
+                                                    description={it.description}
+                                                    imageURL={it.imageUrl}
+                                                    requestsEnabled={this.props.requestsEnabled && this.props.user}
+                                                    totalAvailable={it.totalAvailable}
+                                                    maxRequestQty={it.maxRequestQty}
+                                                    qtyUnreserved={it.qtyUnreserved}
+                                                    category={it.category}
+                                                    key={it.id}
+                                                    id={it.id}
+                                                    addItem={this.props.handleAddItem} // prop that invokes the handleAddItem method of parent container to update its state
+                                                    qtyUpdate={this.props.qtyUpdate} // this prop is the object whose request has been cancelled
+                                                    user={this.props.user}
+                                                />);
+                                            })
+                                            }
+                                        </Item.Group>
+                                    </Accordion.Content>
+                                </div>
+
+                            ))}
+                        </Accordion>);
                 }
-                return normalContent;
+                return (normalContent);
             }}
         </Query>;
 
+        const searchBar = this.props.user ? (
+            <Input type="text"
+                   label="Search items"
+                   name="searchQuery"
+                   onChange={(e, {value}) => {
+                       if (value.length >= 3) {
+                           this.setState({
+                               searchQuery: value.trim().toLowerCase()
+                           });
+                       } else {
+                           this.setState({searchQuery: ""});
+                       }
+                   }
+                   }
+            />) : "";
+
         return (
             <div>
-                <Header as="h1">Inventory</Header>
+                <Grid columns='equal' padded>
```

Comment:
This is probably fine, but have you checked the appearance of this on mobile to make sure the alignment doesn't get weird?

Reply:
Adjusted the column width to be optimized for responsive displays.